In [35]:
# Importing pandas for data loading and manipulation
import pandas as pd

In [36]:
# Loading the raw Titanic training dataset
data = pd.read_csv("../datasets/raw/titanic/train.csv")

data.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


## Separating Features and Target

In supervised machine learning, the dataset is separated into:

- **Features (`X`)**: Input variables used by the model to learn patterns.
- **Target (`y`)**: The output variable that the model tries to predict.

For the Titanic dataset, `Survived` is the target variable:

- `0` = Did not survive
- `1` = Survived

All remaining columns initially form the feature set `X`. Further preprocessing will be applied before model training.

In [37]:
# Separating the input features from the target variable
X = data.drop(columns=['Survived'])
y = data['Survived']

In [38]:
# Cheeckking the saparated features and target values
display(X.head())
print("Target values:")
display(y.head())

,PassengerId,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


Target values:


0    0
1    1
2    1
3    1
4    0
Name: Survived, dtype: int64

In [39]:
#checking the dimensions of the features and target values
print("X shape", X.shape)
print("y shape", y.shape)

X shape (891, 11)
y shape (891,)


In [40]:
# Comparing the dimensions and data types of X and y
print("X type:", type(X))
print("X dimensions:", X.ndim)
print("X shape:", X.shape)

print()

print("y type:", type(y))
print("y dimensions:", y.ndim)
print("y shape:", y.shape)

X type: <class 'pandas.DataFrame'>
X dimensions: 2
X shape: (891, 11)

y type: <class 'pandas.Series'>
y dimensions: 1
y shape: (891,)


### Understanding the Structure of `X` and `y`

`X` is a 2D DataFrame containing the input features, while `y` is a 1D Series containing one target value for each passenger.

In Pandas:

- `data['Survived']` returns a **Series** with shape `(891,)`.
- `data[['Survived']]` returns a **DataFrame** with shape `(891, 1)`.

The values may be the same, but their data structures and dimensions are different.

In [41]:
# Comparing single-bracket and double-bracket column selection
y_series = data['Survived']
y_dataframe = data[['Survived']]

print("Single bracket:")
print("Type:", type(y_series))
print("Shape:", y_series.shape)
print("Dimensions:", y_series.ndim)

print()

print("Double bracket:")
print("Type:", type(y_dataframe))
print("Shape:", y_dataframe.shape)
print("Dimensions:", y_dataframe.ndim)

Single bracket:
Type: <class 'pandas.Series'>
Shape: (891,)
Dimensions: 1

Double bracket:
Type: <class 'pandas.DataFrame'>
Shape: (891, 1)
Dimensions: 2
